### ISSUE TITLE
Masked Label Propagation Example 🚀

### ISSUE THREAD

**Issue by rusty1s** [pyg-team/pytorch_geometric#4040]
### 🚀 The feature, motivation and pitch

[Masked Label Propagation](https://arxiv.org/abs/2009.03509)/Label Re-Use is a popular approach (especially on [OGB datasets](https://ogb.stanford.edu/docs/leader_nodeprop/)) for boosting model performance on transductive tasks. We should definitely provide an example/helper utilities that let PyG users easily use this technique for their own models.

**Comment by Padarn** [comment]
Hey @rusty1s happy to give this a crack. Would you intend to add the `UniMP` network as a model, or just some helpers to use masked label propagation in existing methods?

**Comment by rusty1s** [comment]
This sounds awesome. AFAIK, label re-use is just a short term to refer to masked label propagation. Furthermore, the [`TransformerConv`](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.TransformerConv) already refers to the message passing module of `UniMP`. We should be able to leverage it although masked label propagation should be applicable to any GNN. As such, some helpers + example should be sufficient IMO.

For an example, we can think about trying to replicate the UniMP+`ogbn-arxiv` example. WDYT?

**Comment by Padarn** [comment]
Hey @rusty1s, have been looking around other examples for some inspiration about masking is already happening and wanted to check something:

https://github.com/pyg-team/pytorch_geometric/blob/e18a897e8a63f78288426dc4e1574a9041d8ecae/examples/infomax_inductive.py#L15

In this example the mask is only applied when training the final classifier (inside of test), but the embedding is calculated on the entire dataset.

Is this intended? I can see the logic somewhat but not totally clear.

**Comment by rusty1s** [comment]
That is correct. The `infomax_indutive.py` script represents an unsupervised learning approach, thus all nodes are used during training. I don't think this is necessary for testing though: we could restrict `test_loader` to operate on `data.train_mask | data.val_mask`.

**Comment by rusty1s** [comment]
It's probably worth fixing.

**Comment by Padarn** [comment]
Oh I don't think there is a problem then. The train masking is done inside of the infomax `model.test` - thanks for clarifying

**Comment by Padarn** [comment]
I've built an example in https://github.com/pyg-team/pytorch_geometric/pull/4197

I feel the best way to generalise this would be to provide a utility which generates the per epoc (or per step?) masks for train/test/validation. 

Because we don't know how people will want to actually use the masks (i.e what they will do to zero out their labels - in this MR I just assume the embedding is zero), probably just providing the ability to build the masks is the most general thing we can do?

Could also add a layer that processes the mask and returns the embedding (zero'ed over mask) if better.

**Comment by rusty1s** [comment]
Thank you. Please see my comments in the #4197.
